### PR TITLE
Rename all existing v2 keys so that they are custom text friendly

### PR DESCRIPTION
**Pull request by pkong-ds**
## What's in this PR?
1. rename all existing v2 keys
2. Added `make lint-translation-keys` to `make lint`, which is in CI

## Notes

All conversions in `translation-key-map.json`, which is available until last commit [Remove conversion helper files](https://github.com/authgear/authgear-server/pull/4674/commits/ca71d38cf9c9778a692fc32855be7d1abdbe224e)

<details>

<summary> Outdated </summary>

```

## Pending 

1. sed script to apply these conversions
4. ~apply proper states other than `default`~
EDIT: updated 2

```

</details>

### CODE REVIEW

File: .make-lint-translation-keys-expect [hunk 1]
```diff
@@ -0,0 +1,12 @@
+resources/authgear/templates/en/web/authflowv2/select_account.html:13:24: invalid translation key: "app.name"
+resources/authgear/templates/en/web/authflowv2/__error.html:134:18: translation key not defined: "authflowv2/__error_account"
+resources/authgear/templates/en/web/authflowv2/__error.html:166:20: translation key not defined: "authflowv2/__error_account"
+resources/authgear/templates/en/web/authflowv2/setup_totp.html:34:20: invalid translation key: "copy-button-label"
```
Comment (louischan-oursky) [inline .make-lint-translation-keys-expect]: This is not expected. Introduce a v2 key for this.

File: .make-lint-translation-keys-expect [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+resources/authgear/templates/en/web/authflowv2/view_recovery_code.html:55:20: invalid translation key: "copy-button-label"
```
Comment (louischan-oursky) [inline .make-lint-translation-keys-expect]: This is not expected.

File: .make-lint-translation-keys-expect [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+resources/authgear/templates/en/web/authflowv2/forgot_password.html:109:30: translation key not defined: "__forgot_password_alternative"
+resources/authgear/templates/en/web/authflowv2/forgot_password.html:165:26: translation key not defined: "__forgot_password_alternative"
+resources/authgear/templates/en/web/authflowv2/__error.html:201:31: translation key not defined: "customer-support-link"
+resources/authgear/templates/en/web/authflowv2/__dialog.html:9:14: translation key not defined: "dialog-attr"
+resources/authgear/templates/en/web/authflowv2/__bot_protection_dialog.html:5:19: translation key not defined: "dialog-content"
+resources/authgear/templates/en/web/authflowv2/__bot_protection_dialog.html:4:25: translation key not defined: "dialog-controller-str"
+resources/authgear/templates/en/web/authflowv2/view_recovery_code.html:44:22: invalid translation key: "download-button-label"
```
Comment (louischan-oursky) [inline .make-lint-translation-keys-expect:32]: This is not expected.

File: translation-key-map.json [hunk 4]
```diff
@@ -151,22 +151,22 @@
...
+  "v2-forgot-password-link-sent-description": "v2.page.forgot-password-link-sent.default.description",
+  "v2-forgot-password-link-sent-resend": "v2.page.forgot-password-link-sent.default.resend",
+  "v2-forgot-password-link-sent-resend--unit": "v2.page.forgot-password-link-sent.default.resent-countdown-label-unit",
+  "v2-forgot-password-link-sent-title": "v2.page.forgot-password-link-sent.default.title",
+  "v2-forgot-password-otp-description": "v2.page.forgot-password-otp.default.description",
+  "v2-forgot-password-otp-description--whatsapp": "v2.page.forgot-password-otp.default.whatsapp-description",
+  "v2-forgot-password-otp-page-title": "v2.page.forgot-password-otp.default.title",
+  "v2-forgot-password-send-via-email": "v2.page.forgot-password.default.send-via-email",
+  "v2-forgot-password-send-via-phone": "v2.page.forgot-password.default.send-via-phone",
+  "v2-forgot-password-title": "v2.page.forgot-password.default.title",
```
Comment (louischan-oursky) [inline translation-key-map.json]: This forgot password page should have 2 states `email` and `phone`

File: translation-key-map.json [hunk 5]
```diff
@@ -167,21 +167,21 @@
...
-  "v2-login-id-name": "",
-  "v2-login-link-otp-initial-description": "",
-  "v2-login-link-otp-initial-description-reauth": "",
-  "v2-login-link-otp-invalid-code-description": "",
-  "v2-login-link-otp-matched-description": "",
-  "v2-login-link-otp-matched-title": "",
-  "v2-login-link-otp-page-title": "",
-  "v2-login-link-otp-resend-button-label": "",
-  "v2-login-link-otp-resend-button-label--unit": "",
+  "v2-ldap-branding": "v2.component.ldap-branding.default.label",
```
Comment (louischan-oursky) [inline translation-key-map.json]: This key needs special handling. Let's discuss

File: translation-key-map.json [hunk 6]
```diff
@@ -167,21 +167,21 @@
...
+  "v2-ldap-login-page-description": "v2.page.ldap-login.default.description",
+  "v2-lockout-retry-button-label": "v2.component.lockout.default.retry-button-label",
+  "v2-lockout-retry-button-label--unit": "v2.component.lockout.default.retry-countdown-label-unit",
+  "v2-lockout-subtitle": "v2.component.lockout.default.subtitle",
+  "v2-lockout-title": "v2.component.lockout.default.title",
+  "v2-login-id-name": "v2.error.error-account-not-found-login-id-name",
+  "v2-login-link-otp-initial-description": "v2.page.oob-otp-link.initial.description",
+  "v2-login-link-otp-initial-description-reauth": "v2.page.oob-otp-link.initial-reauth.description",
+  "v2-login-link-otp-invalid-code-description": "v2.page.oob-otp-link.invalid-code.description",
+  "v2-login-link-otp-matched-description": "v2.page.oob-otp-link.matched.description",
```
Comment (louischan-oursky) [inline translation-key-map.json]: `matched` -> `approved`

File: translation-key-map.json [hunk 7]
```diff
@@ -167,21 +167,21 @@
...
+  "v2-login-link-otp-matched-title": "v2.page.oob-otp-link.matched.title",
```
Comment (louischan-oursky) [inline translation-key-map.json]: `matched` -> `approved`

File: translation-key-map.json [hunk 8]
```diff
@@ -182,18 +182,18 @@
...
+  "v2-login-switch-to-signup": "v2.page.login.default.switch-to-signup",
+  "v2-login-title": "v2.page.login.login.default.title",
+  "v2-no-results-found": "v2.component.phone-input.default.no-results-found",
+  "v2-oauth-provider-name": "v2.error.error-account-not-found-oauth-provider-name",
+  "v2-oob-otp-page-resend-button-label": "v2.component.oob-otp-resend-button.default.label",
+  "v2-oob-otp-page-resend-button-label--unit": "v2.component.oob-otp-resend-button.default.countdown-unit",
+  "v2-oob-otp-page-subtitle": "v2.page.enter-oob-otp.default.subtitle",
+  "v2-oob-otp-page-subtitle--whatsapp": "v2.page.enter-oob-otp.default.subtitle-whatsapp",
+  "v2-oob-otp-page-subtitle-reauth": "v2.page.enter-oob-otp.reauth.subtitle",
+  "v2-oob-otp-page-subtitle-reauth--whatsapp": "v2.page.enter-oob-otp.reauth.subtitle-whatsapp",
```
Comment (louischan-oursky) [inline translation-key-map.json]: These subtitles should of 4 states

```
default
auth-whatsapp
reauth
reauth-whatsapp
```

File: translation-key-map.json [hunk 9]
```diff
@@ -225,23 +225,23 @@
...
-  "v2-setup-totp-instead": "",
-  "v2-setup-totp-title": "",
-  "v2-setup-totp-verify-description": "",
-  "v2-setup-totp-verify-rescan-button-label": "",
-  "v2-setup-totp-verify-title": "",
+  "v2-setup-email-otp-code-instead": "v2.component.authflow-branch.default.setup-email-otp-code-instead",
+  "v2-setup-email-otp-link-instead": "v2.component.authflow-branch.default.setup-email-otp-link-instead",
+  "v2-setup-oob-otp-page-subtitle--email": "v2.page.setup-oob-otp.default.subtitle-email",
+  "v2-setup-oob-otp-page-subtitle--sms": "v2.page.setup-oob-otp.default.subtitle-sms",
+  "v2-setup-oob-otp-page-subtitle--whatsapp": "v2.page.setup-oob-otp.default.subtitle-whatsapp",
```
Comment (louischan-oursky) [inline translation-key-map.json]: 3 states

```
email
sms
whatsapp
```

File: .make-lint-translation-keys-expect [hunk 10]
```diff
@@ -1,19 +1,35 @@
...
+resources/authgear/templates/en/web/authflowv2/__csrf_error_page_layout.html:28:18: translation key not defined: "page-content"
+resources/authgear/templates/en/web/authflowv2/__dialog.html:9:14: translation key not defined: "dialog-attr"
 resources/authgear/templates/en/web/authflowv2/__error.html:134:18: translation key not defined: "authflowv2/__error_account"
 resources/authgear/templates/en/web/authflowv2/__error.html:166:20: translation key not defined: "authflowv2/__error_account"
-resources/authgear/templates/en/web/authflowv2/setup_totp.html:34:20: invalid translation key: "copy-button-label"
-resources/authgear/templates/en/web/authflowv2/view_recovery_code.html:55:20: invalid translation key: "copy-button-label"
+resources/authgear/templates/en/web/authflowv2/__error.html:201:31: translation key not defined: "customer-support-link"
+resources/authgear/templates/en/web/authflowv2/__html_head.html:4:19: invalid translation key: "app.name"
+resources/authgear/templates/en/web/authflowv2/__page_frame.html:12:18: translation key not defined: "page-content"
+resources/authgear/templates/en/web/authflowv2/__settings_item.html:4:13: translation key not defined: "some-key"
```
Comment (louischan-oursky) [inline .make-lint-translation-keys-expect:11]: some-key? Huh?

File: .make-lint-translation-keys-expect [hunk 11]
```diff
@@ -1,19 +1,35 @@
...
+resources/authgear/templates/en/web/authflowv2/__settings_page_frame.html:46:18: translation key not defined: "page-content"
+resources/authgear/templates/en/web/authflowv2/__toc_pp_footer.html:23:54: invalid translation key: "terms-of-service-link"
+resources/authgear/templates/en/web/authflowv2/__toc_pp_footer.html:24:53: invalid translation key: "privacy-policy-link"
+resources/authgear/templates/en/web/authflowv2/account_linking.html:94:21: invalid translation key: "printf "v2.component.oauth-branding.default.label-%s" .ProviderType"
+resources/authgear/templates/en/web/authflowv2/enter_password.html:35:11: invalid translation key: "$description_key"
```
Comment (louischan-oursky) [inline .make-lint-translation-keys-expect:16]: Do not seem expected.

File: .make-lint-translation-keys-expect [hunk 12]
```diff
@@ -1,19 +1,35 @@
...
-resources/authgear/templates/en/web/authflowv2/view_recovery_code.html:44:22: invalid translation key: "download-button-label"
-resources/authgear/templates/en/web/authflowv2/__csrf_error_page_layout.html:28:18: translation key not defined: "page-content"
-resources/authgear/templates/en/web/authflowv2/__base_page_frame.html:43:12: translation key not defined: "page-frame-content"
 resources/authgear/templates/en/web/authflowv2/layout.html:5:14: translation key not defined: "widget"
-resources/authgear/templates/en/web/authflowv2/__toc_pp_footer.html:23:54: invalid translation key: "terms-of-service-link"
-resources/authgear/templates/en/web/authflowv2/__toc_pp_footer.html:24:53: invalid translation key: "privacy-policy-link"
-resources/authgear/templates/en/web/authflowv2/__error.html:201:31: translation key not defined: "customer-support-link"
-<REPLACEME> errors found
+resources/authgear/templates/en/web/authflowv2/ldap_login.html:13:11: invalid translation key: "printf "v2.component.ldap-branding.default.label-%s" .server_name"
+resources/authgear/templates/en/web/authflowv2/ldap_login.html:20:11: translation key not defined: "v2.page.ldap-login.default.description"
```
Comment (louischan-oursky) [inline .make-lint-translation-keys-expect:21]: key not defined is not expected.

File: .make-lint-translation-keys-expect [hunk 13]
```diff
@@ -1,19 +1,35 @@
...
+resources/authgear/templates/en/web/authflowv2/login.html:62:18: invalid translation key: "app.name"
+resources/authgear/templates/en/web/authflowv2/login.html:252:25: invalid translation key: "printf "v2.component.oauth-branding.default.label-%s" .provider_type"
+resources/authgear/templates/en/web/authflowv2/login.html:274:25: invalid translation key: "printf "v2.component.ldap-branding.default.label-%s" .server_name"
+resources/authgear/templates/en/web/authflowv2/select_account.html:13:24: invalid translation key: "app.name"
+resources/authgear/templates/en/web/authflowv2/settings_layout.html:3:14: translation key not defined: "widget"
+resources/authgear/templates/en/web/authflowv2/setup_totp.html:34:20: invalid translation key: "copy-button-label"
```
Comment (louischan-oursky) [inline .make-lint-translation-keys-expect:27]: This is unexpected.

File: .make-lint-translation-keys-expect [hunk 14]
```diff
@@ -1,19 +1,35 @@
...
+resources/authgear/templates/en/web/authflowv2/signup.html:40:18: invalid translation key: "app.name"
+resources/authgear/templates/en/web/authflowv2/signup.html:237:23: invalid translation key: "printf "v2.component.oauth-branding.default.label-%s" .provider_type"
+resources/authgear/templates/en/web/authflowv2/signup.html:259:23: invalid translation key: "printf "v2.component.ldap-branding.default.label-%s" .server_name"
+resources/authgear/templates/en/web/authflowv2/verify_login_link.html:4:18: invalid translation key: "app.name"
+resources/authgear/templates/en/web/authflowv2/view_recovery_code.html:44:22: invalid translation key: "download-button-label"
+resources/authgear/templates/en/web/authflowv2/view_recovery_code.html:55:20: invalid translation key: "copy-button-label"
```
Comment (louischan-oursky) [inline .make-lint-translation-keys-expect:33]: This is unexpected.

File: resources/authgear/templates/de/translation.json [hunk 15]
```diff
@@ -17,61 +17,102 @@
...
   "error-rate-limited": "Bitte warten Sie einen Moment, bevor Sie es erneut versuchen.",
+  "v2.error.sms-send-limit-exceeded": "Sie haben das Limit für das Senden von Einmalpasswörtern überschritten. Bitte versuchen Sie es später erneut.",
   "error-sms-send-limit-exceeded": "Sie haben das Limit für das Senden von Einmalpasswörtern überschritten. Bitte versuchen Sie es später erneut.",
+  "v2.error.usage-limit-exceeded": "[An den Entwickler] Ihr Projekt hat das Nutzungslimit überschritten. Bitte überprüfen Sie Ihr Abonnement.",
   "error-usage-limit-exceeded": "[An den Entwickler] Ihr Projekt hat das Nutzungslimit überschritten. Bitte überprüfen Sie Ihr Abonnement.",
+  "v2.error.webhook-disallowed": "Der Vorgang ist nicht zulässig",
   "error-webhook-disallowed": "Der Vorgang ist nicht zulässig",
+  "v2.error.webhook-pre-signup-disallowed": "Die Anmeldung ist nicht zulässig",
   "error-webhook-pre-signup-disallowed": "Die Anmeldung ist nicht zulässig",
-  "error-webhook-disallowed-action": "Zurück",
```
Comment (louischan-oursky) [inline resources/authgear/templates/de/translation.json:73]: Your script is not supposed to remove anything.

**Review by louischan-oursky — CHANGES_REQUESTED** [review]

File: .make-lint-translation-keys-expect (hunk 10 above)
Comment (pkong-ds) [reply .make-lint-translation-keys-expect:11]: Yes, checked it's example usage. Seems our linter is unable to omit `<!-- html comment -->`

```
rg 'some-key'
resources/authgear/templates/en/web/authflowv2/__settings_item.html
4:    "Label" (include "some-key" nil)
```



```
<!-- Example usage:
{{ template "authflowv2/__settings_item.html"
  (dict
    "Label" (include "some-key" nil)
    "Href" "url"
    "MaterialIconName" `email`
    "Children" $.Emails
  )
}}
-->
```

File: .make-lint-translation-keys-expect (hunk 11 above)
Comment (pkong-ds) [reply .make-lint-translation-keys-expect:16]: We do not support `evaluating variables` in linter? ref https://github.com/authgear/authgear-server/blob/main/resources/authgear/templates/en/web/authflowv2/enter_password.html#L35

File: resources/authgear/templates/de/translation.json (hunk 15 above)
Comment (pkong-ds) [reply resources/authgear/templates/de/translation.json:73]: Yes if the key is unused originally. Or do you expect me do open another PR for that?

```
(authgear-main) $ rg 'error-webhook-disallowed-action' ./**/*.html
(authgear-main) $
```

File: translation-key-map.json (hunk 6 above)
Comment (louischan-oursky) [reply translation-key-map.json]:
```
default
approved
invalid
```

File: .make-lint-translation-keys-expect (hunk 11 above)
Comment (louischan-oursky) [reply .make-lint-translation-keys-expect:16]: 1. Edit enter_password.html
  - Delete `"v2-enter-password-page-description"`
  - Delete `"v2-enter-additional-password-page-description"`
  - Eliminate `$description_key`
  - Keep L26-L32 and the `<p>`

File: .make-lint-translation-keys-expect (hunk 12 above)
Comment (louischan-oursky) [reply .make-lint-translation-keys-expect:21]: In ldap_login.html, delete the code.

**Review by louischan-oursky — CHANGES_REQUESTED** [review]
1. No changes in the first part of translation.json
2. All v2. keys are sorted
3. No more v2- keys.

**Comment by pkong-ds** [comment]
@louischan-oursky Updated, thanks 🙏 

Changes starting from [Fix commented node still parsed in translation key linter](https://github.com/authgear/authgear-server/pull/4674/commits/5d50d38d265d377cf4d10676bec153e8b4cae9fc)

Also added `make sort-translations`

**Comment by pkong-ds** [comment]
@louischan-oursky Also [Remove unused v2- translations](https://github.com/authgear/authgear-server/pull/4674/commits/b331d478d7798c9114612476adc585fec49589e2)

File: devtools/translationsorter/compose.go [hunk 16]
```diff
@@ -22,7 +23,8 @@ func ComposeJSON(kvPairs []KeyValuePair, path string) (err error) {
 		if _, err = f.Write([]byte(fmt.Sprintf(lineStart + "  "))); err != nil {
 			return fmt.Errorf("failed to write: %w", err)
 		}
-		byteArray := []byte(fmt.Sprintf("\"%s\": \"%s\"", kvPair.Key, kvPair.Value))
+		value := escapeSpecialChars(kvPair.Value)
+		byteArray := []byte(fmt.Sprintf("\"%s\": \"%s\"", kvPair.Key, value))
```
Comment (louischan-oursky) [inline devtools/translationsorter/compose.go:27]: Actually, you can just use `%q`.

**Review by louischan-oursky — APPROVED** [review]

**Pull request merged**